ocaml/lambda/translcore.ml

```diff
@@ -1009,86 +1044,83 @@ and transl_list_with_shape ~scopes expr_list =
   in
   List.split (List.map transl_with_shape expr_list)
 
-and transl_guard ~scopes guard rhs_sort rhs =
-  let layout = layout_exp rhs_sort rhs in
-  let expr = event_before ~scopes rhs (transl_exp ~scopes rhs_sort rhs) in
-  match guard with
-  | None -> Matching.mk_unguarded_rhs expr
-  | Some (Predicate cond) ->
+and transl_rhs ~scopes rhs_sort rhs =
+  let layout = layout_rhs rhs_sort rhs in
+  let transl_body body =
+    event_before ~scopes body (transl_exp ~scopes rhs_sort body)
```

Comment:
Oof, was this possibly exponential before? I didn't notice that `expr` was translated but then not being used in the pattern-guard case.

In any case, how would you feel about inlining `transl_body` at the places it's used? I think there are just two callsites, and that's not too offensive. Inlining can remove indirection, both semantically and literally.

Reply:
Yeah...kind of an unfortunate oversight, although perhaps it speaks to how having a nicer AST representation of the code is generally better for compiler development.

Inlining sounds good to me, it's a pretty short function body too.